src/scanners/javascript.js

```diff
@@ -33,13 +33,23 @@ export default class JavaScriptScanner {
           message.ruleId = messages.JS_SYNTAX_ERROR.code;
         }
 
+        var code = message.ruleId.toUpperCase();
+        var messageObj = messages[code];
+
+        // Fallback to looking up the message object by the
+        // constant string passed in context.report()
+        if (!messageObj && message.message) {
```

Comment:
I'll make that the only way in a separate patch it'll make for easier review. Can see about using message imports at the same time.